core/common/src/main/java/alluxio/underfs/UnderFileSystem.java

```diff
@@ -301,6 +311,15 @@ public int getValue() {
    */
   String getFingerprint(String path);
 
+  /**
+   * A logical UFS may be composed of one or more physical UFSs. This method is used to determine
+   * the operation mode of the logical UFS based on the physical UFS state.
+   *
+   * @param physicalUfsState the state of all mounted physical UFS
+   * @return the desired operation mode for this UFS
+   */
+  UfsMode getOperationMode(Map<String, UfsMode> physicalUfsState);
```

Comment:
I'm not sure if this API is needed, since owner of the ufs mode state can determine this?

Reply:
the owner can not. the method is introduced so that we manage state only on the master but let the ufs implementation decide which operations are allowed. for instance, for a replicated ufs, if one physical ufs has mode `noAccess`, the operation mode for the logical ufs is `readOnly`

Reply:
Already updated this to a filtered list; I'll update the comment.